src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+  public countrySearchAutocompleteDirective: SkyAutocompleteInputDirective;
+
+  public set selectedCountry(newCountry: SkyCountryFieldCountry) {
+    if (this._selectedCountry !== newCountry) {
+      this._selectedCountry = newCountry;
+
+      this.sortCountriesWithSelectedAndDefault(newCountry);
+
+      this.countrySearchForm.get('countrySearch').setValue(this.selectedCountry);
+
+      if (!this.isFirstChange) {
+        this.onChange(newCountry);
+        this.onTouched();
+
+        this.selectedCountryChange.emit(newCountry);
+      }
+
+      // Do not mark the field as "dirty"
+      // if the field has been initialized with a value.
+      if (this.isFirstChange && this.control) {
+        this.control.markAsPristine();
+      }
+
+      if (this.isFirstChange && newCountry !== null) {
+        this.isFirstChange = false;
+      }
+    }
+  }
+
+  public get selectedCountry(): SkyCountryFieldCountry {
+    return this._selectedCountry;
+  }
+
+  private control: AbstractControl;
+
+  private defaultCountryData: SkyCountryFieldCountry;
+
+  private idle: Subject<any> = new Subject();
+
+  private isFirstChange: boolean = true;
+
+  private _defaultCountry: string;
+
+  private _selectedCountry: SkyCountryFieldCountry;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private formBuilder: FormBuilder,
+    private windowRef: SkyAppWindowRef
+  ) {
+    /**
+     * The json functions here ensures that we get a copy of the array and not the global original.
+     * This ensures that multiple instances of the component don't overwrite the original data.
+     *
+     * We must type the window object as any here as the intl-tel-input library adds its object
+     * to the main window object.
+     */
+    this.countries = JSON.parse(JSON.stringify((window as any)
+      .intlTelInputGlobals.getCountryData()));
+    this.defaultCountryData = this.countries.find(country => country.iso2 === 'us');
+
+    this.countrySearchForm = this.formBuilder.group({
+      countrySearch: new FormControl()
+    });
+
+    if ((<HTMLElement>elRef.nativeElement.parentElement)
+      .classList.contains('sky-phone-field-country-search')) {
+      this.isInPhoneField = true;
+    }
+  }
+
+  public ngOnInit(): void {
+    if (!this.defaultCountry) {
+      this.defaultCountry = 'us';
+    }
+
+    this.countrySearchForm.get('countrySearch').valueChanges.subscribe(newValue => {
+      if (newValue) {
+        this.selectedCountry = newValue;
+      }
+    });
+    this.addEventListeners();
+  }
+
+  public ngOnDestroy(): void {
+    this.selectedCountryChange.complete();
+    this.removeEventListeners();
+  }
+
+  /**
+   * Sets the country to validate against based on the county's iso2 code.
+   * @param countryCode The International Organization for Standardization's two-letter code
+   * for the default country.
+   */
+  public onCountrySelected(newCountry: SkyAutocompleteSelectionChange): void {
```

Comment:
Is this method meant for public use? If not, we can slap an `@internal` on the jsdoc comment. Also, your param should be `newCountry`...not that it matters much.

Reply:
Done